### ISSUE TITLE
Extract audio of current video

### ISSUE THREAD

**Issue by nezuo** [mpv-player/mpv#11157]
### Expected behavior of the wanted feature
An API (like the screenshot input command) that extracts audio given a start and end time and saves it to a file. I want to use this API to extract the audio of a streamed video (no local video on disk).

**Comment by Dudemanguy** [comment]
Why not just pass start and end times to ffmpeg in a script?

**Comment by nezuo** [comment]
I just tried that and if I had to extract an audio clip that starts at 20 minutes, it would take really long time to stream the video up to that point.

**Comment by Dudemanguy** [comment]
I don't understand what you mean. You can invoke the ffmpeg command anytime you want.

**Comment by nezuo** [comment]
It's extracting the correct audio but it has to stream and decode all the way up the 20 minute mark which takes a lot of time. I tested the command by itself, is there a faster way if I did it in an mpv plugin?

**Comment by Dudemanguy** [comment]
It should not take long to "stream and decode all the way up the 20 minute mark". Are you using -ss before your input?

**Comment by nezuo** [comment]
I tried that but it gave me an incorrect result. I want to make sure it's clear that I am not using a local video file, it's a stream URL.

**Comment by Dudemanguy** [comment]
Oh I didn't realize you meant stream as in "internet stream". I'm sure there's some way to do it, but my ffmpeg-foo isn't very strong admittedly. I don't use mpv's encoding mode at all, but it uses ffmpeg internally so I don't see how it would be any better.

**Comment by nezuo** [comment]
That almost works but the dumped video starts before the specified start time. If there was a way to tell when dumped video starts I could maybe extract the exact section I need.

**Comment by guidocella** [comment]
You can't even use `dump-cache` without passing the start time.

**Comment by nezuo** [comment]
I don't know if what I said was confusing, but I did pass in a start time. The resulting file started a little bit before the start time I gave it.

**Comment by guidocella** [comment]
There's no other way to specify the start time for `dump-cache`. You can adjust the start time again with ffmpeg's `-ss` option.

**Comment by nezuo** [comment]
There is no way to know when the new video starts. If I want to cut a video at 20 minutes and it cuts it at 19:54, I don't have a way to know the new video starts at 19:54.

**Comment by christoph-heinrich** [comment]
Maybe it starts dumping at the last keyframe before the specified time. You could dump the whole video and then cut it with ffmpeg. That way you would avoid that inaccuracy.

**Comment by garoto** [comment]
`ffmpeg -ss 00:15:13 -i <a_direct_media_url_ffmpeg_can_play_or_a_yt-dlp_get-url_resolved_url> -t 60 -c:a <copy|libopus|aac> file:shit.<opus|m4a>`

Also this: 
```
> mpv --input-cmdlist| rg ab-loop
ab-loop
ab-loop-dump-cache   filename=String
ab-loop-align-cache
```

**Comment by nezuo** [comment]
@christoph-heinrich It won't really cache the whole video as I watch. If I skip around, it definitely will have holes.

@garoto I've tried every variation of ffmpeg command I could find (including the one you sent) and none of them are exact. I tried ab-loop commands but that still starts at the same incorrect spot.

**Comment by nezuo** [comment]
Would it be a good idea to make the `dump-cache` command more accurate? I don't know C but I could take a crack at it.

**Comment by christoph-heinrich** [comment]
@nezuo Video will always have to be cut a keyframes, but maybe adding a parameter to dump audio only would work.

**Comment by t-8ch** [comment]
There is also `--ao=pcm` that can write audio to a file.

**Comment by nezuo** [comment]
@christoph-heinrich I don't know too much about how videos work but could I cut the video after the fact? I can do that with ffmpeg with local files.

**Comment by christoph-heinrich** [comment]
Yes, but cutting at anything other then keyframes will require re-encoding. I don't know if extracting and cutting audio is possible without re-encoding.

**Comment by nezuo** [comment]
I would be fine with that for my use case.
